synapse/storage/databases/main/__init__.py

```diff
@@ -269,7 +270,8 @@ def get_users_paginate_txn(
 
             sql = f"""
                 SELECT name, user_type, is_guest, admin, deactivated, shadow_banned,
-                displayname, avatar_url, creation_ts * 1000 as creation_ts, approved
+                displayname, avatar_url, creation_ts * 1000 as creation_ts, approved,
+                eu.user_id not null as erased
```

Comment:
On SQLite this is presumably going to be computed as an integer---please explicitly convert this to a bool in Python!

I'm not sure this is legit postgres syntax? https://www.postgresql.org/docs/current/functions-comparison.html doesn't mention it. There is `NOTNULL`, but I think we'd normally spell this as `eu.user_id IS NOT NULL`

Reply:
> When I try this out (via the API endpoint) all the booleans are actually returned as integers: is_guest, admin, deactivated and now erased. It's consisted with all the others, which I assumed would be desirable.

I think we simply didn't realise this was happening (different response types for sqlite/postgres) until after the fact. See e.g. #13505, #13519

> Is it the right time to change them both, or make the whole result consistent while we're at it?

I wouldn't say no to this, but it might be a pain to track things down and clutter up this PR. I'd only worry about the new column you're adding here.